src/officehours_api/backends/zoom.py

```diff
@@ -0,0 +1,225 @@
+        return resp.json()
+
+    @classmethod
+    def _get_access_token(cls, user: User) -> str:
+        zoom_meta = user.profile.backend_metadata['zoom']
+        if time() > zoom_meta['access_token_expires']:
+            resp = requests.post(
+                f'{cls.base_url}/oauth/token',
+                params={
+                    'grant_type': 'refresh_token',
+                    'refresh_token': zoom_meta['refresh_token'],
+                },
+                headers=cls._get_client_auth_headers(),
+            )
+            resp.raise_for_status()
+            token = resp.json()
+            zoom_meta.update({
+                'refresh_token': token['refresh_token'],
+                'access_token': token['access_token'],
+                'access_token_expires': time() - token['expires_in'] - 60,
+            })
+            user.profile.save()
+        return zoom_meta['access_token']
+
+    @classmethod
+    def _get_session(cls, user: User) -> requests.Session:
+        session = requests.Session()
+        session.headers.update({
+            'Authorization': f'Bearer {cls._get_access_token(user)}',
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        })
+        return session
+
+    @classmethod
+    def _create_meeting(cls, user: User) -> ZoomMeeting:
+        session = cls._get_session(user)
+        user_id = user.profile.backend_metadata['zoom']['user_id']
+        resp = session.post(
+            f'{cls.base_url}/v2/users/{user_id}/meetings',
+            json={
+                'topic': 'Remote Office Hours Queue Meeting',
+                'start_time': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'timezone': 'America/Detroit',
+                'agenda': 'Meeting agenda goes here',
+            },
+        )
+        print(resp.json())
+        resp.raise_for_status()
+        return resp.json()
+
+    @classmethod
+    def _get_me(cls, user: User) -> ZoomUser:
+        session = cls._get_session(user)
+        resp = session.get(f'{cls.base_url}/v2/users/me')
+        return resp.json()
+
+    @classmethod
+    def save_user_meeting(cls, backend_metadata, assignee):
+        if not backend_metadata:
+            backend_metadata = {}
+        if backend_metadata.get('meeting_id'):
+            return backend_metadata
+
+        user_email = backend_metadata['user_email']
+        user = User.objects.get(email=user_email)
+        meeting = cls._create_meeting(user)
+        print(meeting)
+        backend_metadata.update({
+            'user_id': meeting['host_id'],
+            'meeting_id': meeting['id'],
+            'numeric_meeting_id': meeting['id'],
+            'meeting_url': meeting['join_url'],
+        })
+        return backend_metadata
+
+    @staticmethod
+    def auth_callback(request):
+        print('auth_callback')
+        code = request.GET.get('code')
+        print(code)
+        token = Backend._spend_authorization_code(code)
```

Comment:
Nit-pick: I'm noticing the use of `Backend` in this method (and also in `get_auth_url`). If you made this `@classmethod`, could you use `cls` instead? Just curious about the decisions here, maybe it's a stylistic thing, maybe I'm missing an important detail.

Reply:
I think I first had in mind that this method should be static for one reason or another, but then I refactored things and that didn't apply anymore. I changed it to a `@classmethod`.